Algorithmia/Data/AlgoDataDirectory.swift

```diff
@@ -8,21 +8,183 @@
 
 import Foundation
 
-class AlgoDirectory:AlgoDataObject {
+typealias AlgoDataListingHandler = ( AlgoDataObject?) -> Void
+
+class AlgoDataDirectory:AlgoDataObject {
+    
     init(client: AlgoAPIClient, dataUrl: String) {
-        super.init(client: client, dataUrl: dataUrl, type: .Directory)
+        var dirUrl = dataUrl
+        if dataUrl.hasSuffix("/") {
+            dirUrl = dataUrl.substring(to: dataUrl.index(before: dataUrl.endIndex))
+        }
+        super.init(client: client, dataUrl: dirUrl, type: .Directory)
+    }
+    
+    func create(_ completion:@escaping AlgoSimpleCompletionHandler) {
+        let entity = try! AlgoJSONEntity(entity: ["name":self.basename()])
+        _ = client.send(method: .POST, path: self.parent()!.getUrl(), data: entity) { (respData, error) in
+            if respData.statusCode == 200 {
+                completion(nil)
+            }
+            else if error == nil {
```

Comment:
Yikes. ok. So if I follow correctly, if data requests are 200, then clients should consider it a success.

(And IMO, we should consider having partially failed deletes return a non-200 response.)